webauthn-rs/src/proto.rs

```diff
@@ -1216,7 +1216,7 @@ fn acd_parser(i: &[u8]) -> nom::IResult<&[u8], AttestedCredentialData> {
     Ok((
         i,
         AttestedCredentialData {
-            aaguid: aaguid.to_vec(),
+            aaguid: aaguid.try_into().expect("took 16 characters"),
```

Comment:
I don't think we should expect here, since this is untrusted user data which could cause a crash. We should map_err and ? here instead.

Reply:
We parsed the aaguid with `take(16usize)(i)`. Shouldn't this have returned an error already if we are unable to get 16?

Reply:
Yes, I think so.